osu.Game/OsuGame.cs

```diff
@@ -84,17 +86,49 @@ private void load()
             if (args?.Length > 0)
             {
                 var paths = args.Where(a => !a.StartsWith(@"-"));
-                ImportBeatmapsAsync(paths);
+                Task.Run(() => BeatmapDatabase.Import(paths));
             }
 
             Dependencies.Cache(this);
 
             PlayMode = LocalConfig.GetBindable<PlayMode>(OsuConfig.PlayMode);
         }
 
-        protected async void ImportBeatmapsAsync(IEnumerable<string> paths)
+        private ScheduledDelegate scoreLoad;
+
+        protected void LoadScore(Score s)
         {
-            await Task.Run(() => BeatmapDatabase.Import(paths));
+            scoreLoad?.Cancel();
+
+            var menu = intro.ChildScreen;
+
+            if (menu == null)
+            {
+                scoreLoad = Schedule(() => LoadScore(s));
+                return;
+            }
+
+            if (!menu.IsCurrentScreen)
+            {
+                menu.MakeCurrent();
+                Delay(500);

```

Comment:
Is the delay just for visuals? If it is required to prevent breakage, then we need to fix the underlying breakage.